src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3315,6 +3318,7 @@ private bool HasTopLevelNullabilityConversion(TypeSymbolWithAnnotations source,
             Debug.Assert(node != null);
             Debug.Assert(operandOpt != null || !operandType.IsNull);
             Debug.Assert(!targetTypeWithNullability.IsNull);
+            Debug.Assert((object)target != null || assignmentKind != AssignmentKind.Argument);
```

Comment:
Is there really no way for an AssignmentKind.Return to get here with no Target? While we don't have any tests that are triggering this, it seems like a return statement with a conversion should get here.

Reply:
Sorry, scratch this, misread the assert.

---
In reply to: [241615041](https://github.com/dotnet/roslyn/pull/31771#discussion_r241615041) [](ancestors = 241615041)